plugins/woocommerce/src/Internal/Admin/Notes/WooSubscriptionsNotes.php

```diff
@@ -221,6 +221,9 @@ public static function get_note() {
 	 * @return int|false
 	 */
 	public function get_product_id_from_subscription_note( &$note ) {
+		if ( false === $note ) {
```

Comment:
An even more robust fix would be to check if `! is_object( $note )`.

Reply:
Thanks, I updated it here: https://github.com/woocommerce/woocommerce/pull/46472/commits/5b8ebb5b159927a990bdd90b67f2203d6b343ab9